File: How_To_Program_Exercises/Visual_C#_How_To_Program2012_Fifth/Chapter4/InvoiceClass/Invoice.cs
//
//	Invoice Class
//

using System;

class Invoice
{
	private string number;
	private string description;
	private decimal pricePerItem;
	private int quantity;
	
	public Invoice ( string n, string d, decimal p, int q)
	{
		Number = n;
		Description = d;
		PricePerItem = p;
		Quantity = q;
	}

	public string Number
	{
		get { return number; }
		set { number = value; }
	}
	
	public string Description
	{
		get { return description; }
		set { description  = value; }
	}
	
	public decimal PricePerItem 
	{
		get { return pricePerItem; }
		set 
		{
			if ( value >= 0.0M )
				pricePerItem = value; 
		}
	}
	
	public int Quantity
	{
		get { return quantity; }
		set 
		{
			if ( quantity >= 0 )
				quantity = value; 
		}
	}

	public decimal GetInvoiceAmount ()
	{
		return PricePerItem * Quantity;
	}
}

File: How_To_Program_Exercises/JavaHowToProgram9/Chapter3/InvoiceClass/Invoice.java
//
//	Invoice Class 
//

public class Invoice
{
	private String number;
	private String description;
	private int quantity;
	private double pricePerItem;
	
	public Invoice (String n, String d, int q, double p)
	{
		number = n;
		description = d;
		
		if ( q > 0 )
			quantity = q;
			
		if ( p > 0.0 )
		pricePerItem = p;
	}
	
	public String getNumber ()
	{
		return number;
	}
	
	public void setNumber (String n)
	{
		number = n;
	}
	
	public String getDescription ()
	{
		return description;
	}
	
	public void setDescription (String d)
	{
		description = d;
	}
	
	public int getQuantity ()
	{
		return quantity;
	}
	
	public void setQuantity (int q)
	{
		if ( q >= 0)
			quantity = q;
		else
			quantity = 0;
	}
	
	public double getPricePerItem ()
	{
		return pricePerItem;
	}
	
	public void setPricePerItem (double p)
	{
		if ( p >= 0.0)
			pricePerItem = p;
		else	
			pricePerItem = 0.0;
	}

	public double getInvoiceAmount ()
	{
		return pricePerItem * quantity;
	}
}